client/engine/messageservice/p2p-message-service/service.go

```diff
@@ -179,3 +173,35 @@ func (s *P2PMessageService) Close() error {
 	s.p2pHost.RemoveStreamHandler(PROTOCOL_ID)
 	return s.p2pHost.Close()
 }
+
+// generateKey generates a ECDSA key deterministically from the given address
+// TOODO: This is probably a security hole, but it's good enough for now
```

Comment:
This makes it pretty trivial to generate the message key if you know the address. For our use case this is fine (since we don't even care if the message traffic is encrypted, since we verify signatures about things we care about).

Still, probably better to move this to test code and just have the message service accept an `ecdsa.PrivKey` for it's message service. In the real world we can generate one randomly but in our tests we can use a deterministic set of keys.

Reply:
It seems like, before this PR, we are generating the p2p message key from the state channels private key. Is there a reason we are moving away from this approach?

Reply:
> It seems like, before this PR, we are generating the p2p message key from the state channels private key. Is there a reason we are moving away from this approach?

FYI, I ended up tackling some of this in a separate [PR](https://github.com/statechannels/go-nitro/pull/1194), so this function is outdated. In that PR `generateMessageKey` accepts a `pk` not the address.